frontend/fontlist.lua

```diff
@@ -74,13 +74,13 @@ local function getExternalFontDir()
 end
 
 local function _readList(target, dir)
-    -- lfs.dir non-exsitent directory will give error, weird!
+    -- lfs.dir non-existent directory will give an error, weird!
     local ok, iter, dir_obj = pcall(lfs.dir, dir)
     if not ok then return end
     for f in iter, dir_obj do
         if lfs.attributes(dir.."/"..f, "mode") == "directory" and f ~= "." and f ~= ".." then
             _readList(target, dir.."/"..f)
-        else
+        elseif lfs.attributes(dir.."/"..f, "mode") == "file" then
```

Comment:
@NiLuJe Actually on this note, on newer Android Droid is Roboto, so it's not just theory: 

```
lrw-r--r-- root     root              2017-07-12 21:51 DroidSans-Bold.ttf -> Roboto-Bold.ttf
lrw-r--r-- root     root              2017-07-12 21:51 DroidSans.ttf -> Roboto-Regular.ttf
```

(But I don't think

Reply:
Where by "newer" I presumably mean 4.0 and up. (I can only test 2.3 and 4.1 and up because the 4.0 emulator has always been borked.)

Reply:
Ah, nice find ;).

I'm wondering if there are any other possible values for attributes besides dir/file/link? (In which case the original test was perfectly fine ;p).

Reply:
https://keplerproject.github.io/luafilesystem/manual.html
> string representing the associated protection mode (the values could be file, directory, link, socket, named pipe, char device, block device or other)

Reply:
Ah, thanks, it always takes me forever to grok the lfs docs ;).

Reply:
So I don't think any of the others should be relevant?

Reply:
Yup! :+1: